File: app/src/main/java/com/example/shiranpeer/burritolocator/service/PlacesService.java
package com.example.shiranpeer.burritolocator.service;

import com.example.shiranpeer.burritolocator.model.PlacesResult;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface PlacesService {

    @GET("place/nearbysearch/json?")
    Observable<PlacesResult> fetchNearbyPlaces(@Query(value = "types", encoded = true) String type, @Query(value = "location",
            encoded = true) String location, @Query(value = "keyword", encoded = true) String keyword,
                                               @Query(value = "rankby", encoded = true) String rankBy,
                                               @Query(value = "key", encoded = true) String key);

    @GET("place/nearbysearch/json?")
    Observable<PlacesResult> fetchNearbyPlacesAdditionalResults(@Query(value = "pagetoken", encoded = true) String pageToken,
                                                                @Query(value = "key", encoded = true) String key);

}
  

File: app/src/main/java/com/example/shiranpeer/burritolocator/service/PlacesService.kt
package com.example.shiranpeer.burritolocator.service

import com.example.shiranpeer.burritolocator.model.PlacesResult

import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query

interface PlacesService {

    @GET("place/nearbysearch/json?")
    fun fetchNearbyPlaces(@Query(value = "types", encoded = true) type: String, @Query(value = "location", encoded = true)
    location: String, @Query(value = "keyword", encoded = true) keyword: String,
                          @Query(value = "rankby", encoded = true) rankBy: String,
                          @Query(value = "key", encoded = true) key: String): Observable<PlacesResult>

    @GET("place/nearbysearch/json?")
    fun fetchNearbyPlacesAdditionalResults(@Query(value = "pagetoken", encoded = true) pageToken: String,
                                           @Query(value = "key", encoded = true) key: String): Observable<PlacesResult>

}
  